v8: add missing ',' in OpenBSD's 'sources' section

qbit opened the pull request:
Fix issues #15784 and https://github.com/nodejs/help/issues/992

##### Checklist

- [X] `make` (OpenBSD)


##### Affected core subsystem(s)
v8

---

I can run a diff [upstream](https://github.com/v8/v8/blob/master/gypfiles/v8.gyp#L2066) as well - but IMO it would be better if someone like @bnoordhuis did, as I don't know who does the reviews in v8 :D

addaleax commented:
> I can run a diff upstream as well - but IMO it would be better if someone like @bnoordhuis did, as I don't know who does the reviews in v8 :D

Fwiw, the guide at https://github.com/v8/v8/wiki/Contributing has become pretty usable.

I’m not sure whether this can actually land upstream though, because they are migrating (have migrated?) away from gyp anyway and the file is not in upstream `src/` … @hashseed what should we do here?

qbit commented:
Nice! The file is still in the tree : https://github.com/v8/v8/blob/master/gypfiles/v8.gyp#L2066

hashseed commented:
Don't expect gyp files to make it into the next V8 release. That being said, I would like to ensure that gyp files are in a good state before removing them from V8 and handing them over to Node.js. Since the file moved, floating this patch would be somewhat annoying.

I took the liberty of [fixing this upstream](https://chromium-review.googlesource.com/#/c/v8/v8/+/893261).

qbit commented:
Crud - I spoke too soon (committed too soon? :P)!  Missed this in my diff:

```
diff --git a/deps/v8/src/v8.gyp b/deps/v8/src/v8.gyp
index 769078479e..a7ce858022 100644
--- a/deps/v8/src/v8.gyp
+++ b/deps/v8/src/v8.gyp
@@ -2068,6 +2068,7 @@
                 '-L/usr/local/lib -lexecinfo',
             ]},
             'sources': [
+              'base/debug/stack_trace_posix.cc',
               'base/platform/platform-openbsd.cc',
               'base/platform/platform-posix.h',
               'base/platform/platform-posix.cc',
```
I will re-run my :test before pushing a fix - can we hold of on merging this for a little bit?

hashseed commented:
Updated the upstream patch.

qbit commented:
OK, updated the pr to include `stack_trace_posix.cc`. I also added another commit to add a `-I` to cflags in `common.gyp`.. I can remove that and make another PR if needed.

hashseed commented:
[Upstream patch](https://chromium-review.googlesource.com/c/v8/v8/+/893261) also merged.

BridgeAR commented:
CI https://ci.nodejs.org/job/node-test-pull-request/12885/

jasnell commented:
Landed in eb82b21 and 68d9f63

Alhadis commented:
Thanks so much for addressing this, guys! :heart: Built like a charm on OpenBSD 6.1.

@hashseed I'm attempting to compile a standalone binary of V8 itself (mainly for debugging and examining disassembler output), but the [`depot_tools`](https://www.chromium.org/developers/how-tos/install-depot-tools) raises an error during setup:

~~~
Traceback (most recent call last):
	File "/home/Alhadis/Mirrors/depot_tools/gclient.py", line 2857, in <module>
		sys.exit(main(sys.argv[1:]))
	File "/home/Alhadis/Mirrors/depot_tools/gclient.py", line 2843, in main
		return dispatcher.execute(OptionParser(), argv)
	File "/home/Alhadis/Mirrors/depot_tools/subcommand.py", line 252, in execute
		return command(parser, args[1:])
	File "/home/Alhadis/Mirrors/depot_tools/gclient.py", line 2598, in CMDsync
		ret = client.RunOnDeps('update', args)
	File "/home/Alhadis/Mirrors/depot_tools/gclient.py", line 1584, in RunOnDeps
		work_queue.flush(revision_overrides, command, args, options=self._options)
	File "/home/Alhadis/Mirrors/depot_tools/gclient_utils.py", line 1075, in run
		self.item.run(*self.args, **self.kwargs)
	File "/home/Alhadis/Mirrors/depot_tools/gclient.py", line 954, in run
		self.ParseDepsFile()
	File "/home/Alhadis/Mirrors/depot_tools/gclient.py", line 772, in ParseDepsFile
		for key, value in self.parent.get_vars().iteritems():
	File "/home/Alhadis/Mirrors/depot_tools/gclient.py", line 1277, in get_vars
		'host_os': _detect_host_os(),
	File "/home/Alhadis/Mirrors/depot_tools/gclient.py", line 1295, in _detect_host_os
		return _PLATFORM_MAPPING[sys.platform]
KeyError: 'openbsd6'
~~~

Line 1295 in `depot_tools/gclient.py` points to this:

~~~python
_PLATFORM_MAPPING = {
  'cygwin': 'win',
  'darwin': 'mac',
  'linux2': 'linux',
  'win32': 'win',
  'aix6': 'aix',
}
~~~

Tampering with the platform mapping by adding `'openbsd6': 'bsd'` in there only raised more problems with getting the build to start...

Could this be amended to accommodate OpenBSD users? Or is it possible to build V8 without the `depot_tools`? V8's wiki-page on [*"Building from Source"*](https://github.com/v8/v8/wiki/Building-from-Source) makes that sound like a bad idea:

> **Don’t simply `git clone` the V8 repository!**

qbit commented:
There are a few examples of building t stand-alone via the ports tree (lang/libv8 - and iirc I sent a few updated versions). The current stand-alone version is fairly old, but with the patches in that or, you should be able to build more recent versions (without depottools)

MylesBorins commented:
This landed cleanly on v8.x. Opted to not land on v6.x, let me know if that was incorrect

qbit commented:
That's fine, OpenBSD is now on the v8.x line now.. and v6 wasn't impacted.
